### PR TITLE
Docusaurus Update and CD workflows

### PR DESCRIPTION
**Pull request by TheRootOf3**
This PR introduces the following:
- Updated Docusaurus version from 2.4.1 to 2.4.3
- Added CD workflows for automated deployments of `main` branch updates (used default workflows from [official Docusaurus docs](https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions).

The website builds.
image

### CODE REVIEW

File: .github/workflows/test-deploy.yml
Comment (Willmish) [inline .github/workflows/test-deploy.yml:1]: Why do we need 2 GH Actions? what is the test-deploy.yml for? it seems to do the same athing as the first half of deploy.yml

**Review by Willmish — CHANGES_REQUESTED** [review]
See questions below

File: .github/workflows/test-deploy.yml
Comment (TheRootOf3) [reply .github/workflows/test-deploy.yml:1]: Hey, `test-deploy` has a different trigger. It pre-builds the website on pull requests to `main` (`deploy` on pushes to `main`), ensuring it builds correctly without an actual deployment.

Comment (Willmish) [reply .github/workflows/test-deploy.yml:1]: ah missed the different triggers. sounds good!

**Review by Willmish — APPROVED** [review]
LGTM

**Pull request merged**